File: CS/AuthenticationOwin.Web/LogonTemplateContent1.ascx.cs
using System;
using System.Collections.Generic;
using DevExpress.ExpressApp.Templates;
using DevExpress.ExpressApp.Web.Controls;
using DevExpress.ExpressApp.Web.Templates;
namespace AuthenticationOwin.Web
{
    public partial class LogonTemplateContent1 : TemplateContent, IXafPopupWindowControlContainer, IHeaderImageControlContainer
    {
        public override void SetStatus(ICollection<string> statusMessages)
        {
        }
        public override IActionContainer DefaultContainer
        {
            get { return null; }
        }
        public override object ViewSiteControl
        {
            get { return viewSiteControl; }
        }
        public XafPopupWindowControl XafPopupWindowControl
        {
            get { return PopupWindowControl; }
        }
        public override void BeginUpdate()
        {
            base.BeginUpdate();
            this.PopupActions.BeginUpdate();
        }
        public override void EndUpdate()
        {
            this.PopupActions.EndUpdate();
            base.EndUpdate();
        }
        public ThemedImageControl HeaderImageControl { get { return TIC; } }
        protected override void OnLoad(EventArgs e) {
            base.OnLoad(e);
            OAuthActions.Menu.ItemWrap = true;
        }
    }
}

File: VB/AuthenticationOwin.Web/LogonTemplateContent1.ascx.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports DevExpress.ExpressApp.Templates
Imports DevExpress.ExpressApp.Web.Controls
Imports DevExpress.ExpressApp.Web.Templates
Namespace AuthenticationOwin.Web
	Partial Public Class LogonTemplateContent1
		Inherits TemplateContent
		Implements IXafPopupWindowControlContainer, IHeaderImageControlContainer
		Public Overrides Sub SetStatus(ByVal statusMessages As ICollection(Of String))
		End Sub
		Public Overrides ReadOnly Property DefaultContainer() As IActionContainer
			Get
				Return Nothing
			End Get
		End Property
		Public Overrides ReadOnly Property ViewSiteControl() As Object
			Get
				Return viewSiteControl_Renamed
			End Get
		End Property
		Public ReadOnly Property XafPopupWindowControl() As XafPopupWindowControl Implements IXafPopupWindowControlContainer.XafPopupWindowControl
			Get
				Return PopupWindowControl
			End Get
		End Property
		Public Overrides Sub BeginUpdate()
			MyBase.BeginUpdate()
			Me.PopupActions.BeginUpdate()
		End Sub
		Public Overrides Sub EndUpdate()
			Me.PopupActions.EndUpdate()
			MyBase.EndUpdate()
		End Sub
		Public ReadOnly Property HeaderImageControl() As ThemedImageControl Implements IHeaderImageControlContainer.HeaderImageControl
			Get
				Return TIC
			End Get
		End Property
		Protected Overrides Sub OnLoad(ByVal e As EventArgs)
			MyBase.OnLoad(e)
			OAuthActions.Menu.ItemWrap = True
		End Sub
	End Class
End Namespace
